go/digest/digest_test.go

```diff
@@ -65,25 +61,74 @@ func TestValidateDigests_Errors(t *testing.T) {
 func Test_New(t *testing.T) {
 	t.Parallel()
 	testcases := []struct {
-		label      string
-		hash       string
-		size       int64
-		wantDigest *repb.Digest
-		wantErr    bool
+		label   string
+		digest  Digest
+		wantErr bool
 	}{
-		{"SHA256", dSHA256.Hash, dSHA256.SizeBytes, dSHA256, false},
-		{"Invalid", dInvalid.Hash, dInvalid.SizeBytes, nil, true},
+		{"SHA256", dSHA256, false},
+		{"Invalid", dInvalid, true},
 	}
 	for _, tc := range testcases {
-		gotDigest, err := New(tc.hash, tc.size)
+		gotDigest, err := New(tc.digest.Hash, tc.digest.Size)
 		if err == nil && tc.wantErr {
-			t.Errorf("%s: New(%s, %d) = (_, nil), want (_, error)", tc.label, tc.hash, tc.size)
+			t.Errorf("%s: New(%s, %d) = (_, nil), want (_, error)", tc.label, tc.digest.Hash, tc.digest.Size)
+			continue
 		}
-		if err != nil && !tc.wantErr {
-			t.Errorf("%s: New(%s, %d) = (_, %v), want (_, nil)", tc.label, tc.hash, tc.size, err)
+		if err != nil {
+			if !tc.wantErr {
+				t.Errorf("%s: New(%s, %d) = (_, %v), want (_, nil)", tc.label, tc.digest.Hash, tc.digest.Size, err)
+			}
+			continue
 		}
-		if diff := cmp.Diff(tc.wantDigest, gotDigest); diff != "" {
-			t.Errorf("%s: New(%s, %d) = (%v, _), want (%v, _), diff (-want +got):\n:%s", tc.label, tc.hash, tc.size, gotDigest, tc.wantDigest, diff)
+		if tc.digest != gotDigest {
+			t.Errorf("%s: New(%s, %d) = (%v, _), want (%v, _)", tc.label, tc.digest.Hash, tc.digest.Size, gotDigest, tc.digest)
+		}
+	}
+}
+
+func Test_NewFromProto(t *testing.T) {
+	t.Parallel()
+	testcases := []struct {
+		label   string
+		digest  Digest
+		wantErr bool
+	}{
+		{"SHA256", dSHA256, false},
+		{"Invalid", dInvalid, true},
+	}
+	for _, tc := range testcases {
+		dPb := tc.digest.ToProto()
+		gotDigest, err := NewFromProto(dPb)
+		if err == nil && tc.wantErr {
+			t.Errorf("%s: NewFromProto(%v) = (_, nil), want (_, error)", tc.label, dPb)
+			continue
+		}
+		if err != nil {
+			if !tc.wantErr {
```

Comment:
I think its generally advised to have a separate test for error cases to avoid this code here.

Reply:
Good idea. Done.